modules/core/src/main/java/org/apache/ignite/internal/util/VersatileReadWriteLock.java

```diff
@@ -338,15 +338,15 @@ public boolean tryWriteLock(long timeout, TimeUnit unit) throws InterruptedExcep
      * @throws IllegalMonitorStateException thrown if the write lock is not acquired.
      */
     public void writeUnlock() {
-        int curState = state;
-        // There could still be some read locks if the write lock was taken forcefully.
-        int readLocks = readLocks(curState);
+        while (true) {
+            int curState = state;
+            // There could still be some read locks if the write lock was taken forcefully.
+            int readLocks = readLocks(curState);
```

Comment:
Why do we get the read locks before the `if ` condition below?

Reply:
Moved the check before getting read locks count